src/services/BaseCacheService.ts

```diff
@@ -1,17 +1,43 @@
 import { BaseCacheResource } from '../resources/BaseCacheResource'
-
+const ttl: number = 3_600_000 // 1 hour
 export abstract class BaseCacheService<T> {
-  private cacheResource
 
+
+  private cacheResource
+  private data = null
+  private updatedAt: number = 0
   constructor() {
     this.cacheResource = new BaseCacheResource<T>()
   }
 
   public async getAll() {
+
     throw new Error('Not implemented')
   }
 
   public async get(id: string) {
+    const hasData = Boolean(this.data)
+    const isFresh = Date.now() - this.updatedAt < ttl
```

Comment:
Checking this.updatedAt might not be the best solution, think about it like this

GET api/contacts happens at time stamp 10:00

GET api/contacts?filter=parth at time stamp 11:00

now, your this.updatedAt would be at 11:00, and the stale time is 1 hour

say we perform GET api/contacts again, at 11:10, isFresh would be true, even though the query took place a long time before your staletime limit